### PR TITLE
img_proof : move logic waiting for registration into our code

### PR DESCRIPTION
**Pull request by asmorodskyi**

### CODE REVIEW

**Comment by asmorodskyi** [comment]
VR : http://autobot.qa.suse.de/tests/2057

File: tests/publiccloud/img_proof.pm [hunk 1]
```diff
@@ -15,35 +15,35 @@ use Mojo::Base 'publiccloud::basetest';
...
+our $test_sles_on_demand_for_dev = 'test_sles_wait_on_registration,test_refresh,test_sles_smt_reg,test_sles_guestregister';
 
 our $azure_byos_updates      = 'test_sles,test_sles_azure';
-our $azure_on_demand_updates = 'test_sles_wait_on_registration,test_sles,test_sles_on_demand,test_sles_azure';
+our $azure_on_demand_updates = 'test_sles,test_sles_on_demand,test_sles_azure';
 
-our $azure_byos      = 'test_sles_azure' . $test_sles_for_dev;
-our $azure_on_demand = 'test_sles_wait_on_registration,test_sles_azure' . $test_sles_on_demand_for_dev;
+our $azure_byos      = $test_sles_for_dev . ',test_sles_azure';
+our $azure_on_demand = $test_sles_for_dev . ',' . $test_sles_on_demand_for_dev . ',test_sles_azure';
```
Comment (b10n1k) [inline tests/publiccloud/img_proof.pm:33]: Can we create a sub to wrap the format here.
something like 
```
sub formatter {
my @tests = @_;
return join ",", @tests;
}
our $azure_on_demand = formatter $test_sles_for_dev,$test_sles_on_demand_for_dev,test_sles_azure;
```
wdyt?

Comment (asmorodskyi) [reply tests/publiccloud/img_proof.pm:33]: sorry if I misunderstood you but for it sounds like you trying to suggest something what we already has https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/provider.pm#L187

**Comment by asmorodskyi** [comment]
> VR : http://autobot.qa.suse.de/tests/2057

all jobs in same host up to 2071 they all failed but in "normal" way due to failed img_proof tests . same  way as it currently failing in osd

**Comment by asmorodskyi** [comment]
so from my POV this PR is ready to be merged

**Pull request merged**